src/Redux/Book/bookSlice.js

```diff
@@ -0,0 +1,41 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  books: [
+    {
+      item_id: 'item1',
+      title: 'The Great Gatsby',
+      author: 'John Smith',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
+};
+
+const bookSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.books.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      const itemId = action.payload;
+      state.books = state.books.filter((book) => book.item_id !== itemId);
+    },
+  },
+});
```

Comment:
- [ ] [OPTIONAL] One of the backbone and peculiarities of redux is not to mutate the state directly but rather using pure functions to update your store,

To make the Redux slice pure, you should avoid directly mutating the state inside the reducer functions. Here's how you can modify your code to make it pure:

```
const bookSlice = createSlice({
  name: 'book',
  initialState,
  reducers: {
    addBook: (state, action) => {
      return {
        ...state,
        books: [...state.books, action.payload],
      };
    },
    removeBook: (state, action) => {
      const itemId = action.payload;
      return {
        ...state,
        books: state.books.filter((book) => book.item_id !== itemId),
      };
    },
  },
});
```